### PR TITLE
chore: conformance profiles API version and channel

### PR DESCRIPTION
**Pull request by mlavacca**
**What type of PR is this?**

/kind gep
/area conformance

**What this PR does / why we need it**:

In support of https://github.com/kubernetes-sigs/gateway-api/issues/1709, the GEP-1709 has been updated with content on how to detect the `gatewayAPIVersion` and the `gatewayAPIChannel` and how to check their consistency.

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

**Review by shaneutt — APPROVED** [review]
/approve
/hold to allow others to take a look
/cc @sunjayBhatia @arkodg @robscott

**Review by arkodg — APPROVED** [review]

**Comment by shaneutt** [comment]
test suite version tracking is a good point.

**Comment by shaneutt** [comment]
> one issue I could see arising is how to handle an outdated set of CRDs in the cluster in comparison w/ the test suite (e.g. v0.X.Y CRDs vs. running the v1.A.B tests)
> 
> I don't expect too much in the way of behavior _change_ between versions going forward since the API is stabilizing, but things that are considered experimental in older versions may no longer be as the tests/versions progress, which could maybe cause a test failure/reporting issue
> 
> I wonder if in addition to this we should consider writing in the report what version the _test suite_ is from and discouraging too much version skew (or maybe not allowing any for certified reports)

Test suite version tracking is a good point. I think we could incorporate that here, unless you'd like to do a follow up PR to add this and any other thoughts you have on the proposal so far @sunjayBhatia?

**Comment by sunjayBhatia** [comment]
> Test suite version tracking is a good point. I think we could incorporate that here, unless you'd like to do a follow up PR to add this and any other thoughts you have on the proposal so far @sunjayBhatia?

If @mlavacca wants to add it here that's great, otherwise I'm happy to make a follow-up 👍🏽

**Comment by mlavacca** [comment]
> If @mlavacca wants to add it here that's great, otherwise I'm happy to make a follow-up 👍🏽

Thanks for this good point @sunjayBhatia, I'll address it in this PR.

**Review by sunjayBhatia — APPROVED** [review]
looks great @mlavacca!

**Comment by shaneutt** [comment]
Thanks for the review @arkodg and @sunjayBhatia!

I'll let one of my co-maintainers drop the final LGTM:

/cc @robscott @youngnick

File: geps/gep-1709.md
Comment (robscott) [inline geps/gep-1709.md:321]: Does this mean that most implementations would need to submit separate reports for each Gateway API Channel? I'm not sure we gain any value from that since the experimental channel is always a superset of the standard channel, but I may be missing something here.

I agree that the version of tests that are being run is critical though, so I'm glad that's included here.

File: geps/gep-1709.md
Comment (robscott) [inline geps/gep-1709.md:277]: What happens when there's some skew here? For example, most of our patch releases just include new conformance tests without meaningful changes to the API spec. Would these not run if the CRD version was v0.7.0 and the test version was either v0.7.1 or v0.7.0-dev?

File: geps/gep-1709.md
Comment (robscott) [inline geps/gep-1709.md:288]: Tiny nit: The line lengths feel a bit weird here, I can't tell if 288 is intended to start a new paragraph, but it is not doing that when the markdown is rendered.

**Review by robscott — COMMENTED** [review]
Thanks @mlavacca! Mostly LGTM, just some questions.

File: geps/gep-1709.md
Comment (mlavacca) [reply geps/gep-1709.md:321]: Since the experimental channel is a superset of the standard one, we could treat the report in the same way, which means being conformant to the experimental channel implies being conformant to the standard one. So that you just need to submit the experimental report to declare the standard conformance as well. WDYT?

File: geps/gep-1709.md
Comment (mlavacca) [reply geps/gep-1709.md:277]: That makes sense, we could allow a skew within the same minor version and block conformance where the CRDs and the suite have different minor versions.

Comment (shaneutt) [reply geps/gep-1709.md:277]: Is this a situation where we should start as strict as possible, and ease off _if_ we get complaints?

File: geps/gep-1709.md
Comment (robscott) [reply geps/gep-1709.md:321]: That's kinda what I'm thinking - we may just want to recommend that implementations that support experimental features run conformance tests with experimental CRDs. I'm not sure that we need to actually state that in the report, but don't feel strongly either way.

File: geps/gep-1709.md
Comment (robscott) [reply geps/gep-1709.md:277]: I think that's also fine. I guess this doesn't prevent running conformance tests, just generating a conformance report, right? I think it's fine to limit conformance reports to tightly matching versions, but I don't want to prevent running the latest set of conformance tests based on this.

Comment (mlavacca) [reply geps/gep-1709.md:277]: Yep, running conformance tests won't be affected, only generating reports.

**Comment by robscott** [comment]
Thanks @mlavacca!

/lgtm

**Pull request merged**
